### ISSUE TITLE
Document cache 

### ISSUE THREAD

**Issue by kal-el11** [eclipse/lemminx#1611]
Does the xml language server hold document model of all the xml files in a workspace/project (including those which has not been opened in the editor)?

**Comment by angelozerr** [comment]
No, today only opened document are cached as DOMDocument. The other cache used is for XSD, DTD, RelaxNG (Xerces gramar) when XML reference a grammar the grammar is cached to avoid parse it when XML isvalidated against XS, DTD.

I think it should be nice to manage closed document to scan all XML files from a workspace/project and report all errors of all XML files even if files are closed, but it requires some work.

Why do you need some information about document cache?

**Comment by kal-el11** [comment]
I have a requirement to analyze and build the data model of all the xml files in my project and keep my data model updated when the xml files encounters a change.
